src/breakpoints.jl

```diff
@@ -0,0 +1,186 @@
+module Breakpoints
+
+using ..JuliaInterpreter
+using JuliaInterpreter: JuliaFrameCode, JuliaStackFrame,
+                        prepare_framecode, framedict, get_source, sparam_syms,
+                        linenumber
+
+export breakpoint
+
+# A type that is unique to this package for which there are no valid operations
+struct Unassigned end
+
+"""
+    Breakpoint(framecode, stmtidx)
+
+A reference to a breakpoint at a particular statement index `stmtidx` in `framecode`.
+"""
+struct Breakpoint
+    framecode::JuliaFrameCode
+    stmtidx::Int
+end
+
+function Base.show(io::IO, bp::Breakpoint)
+    lineno = linenumber(bp.framecode, bp.stmtidx)
+    print(io, "breakpoint(", bp.framecode.scope, ", ", lineno, ')')
+end
+
+const _breakpoints = Set{Breakpoint}()
```

Comment:
I think this should be a vector and the API in the REPL debugger for disabling or removing a breakpoint would give the index into that vector. Also when showing it is nice to do it in the order that they where added.

Reply:
https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/81/files#r261159550 applies here as well

Reply:
I think it'd also be useful to have an API for accessing this (probably readonly).

Reply:
Is the main purpose to list the current breakpoints? Or are there other uses intended?

Unless I'm missing something, making it readonly seems very tricky to me. I'd favor a `print_breakpoints` function that just dumps textual information.

Reply:
Oh, now I understand: you want to disable a specific breakpoint by providing its index. Gotcha.

Reply:
Like in gdb you do `info b` and then you can disable the breakpoint based on an index.

Reply:
I meant "readonly" in the sense that mutating the vector returned by `get_breakpoints` (or whatever) isn't supported/encouraged by the API. Instead we'd recommend using the `breakpoint()`/`remove()` API.